webrender/build.rs

```diff
@@ -27,16 +341,374 @@ fn write_shaders(glsl_files: Vec<PathBuf>, shader_file_path: &Path) {
+                ];
+            } else if file_name.starts_with("debug_font") {
+                descriptors = vec![
+                    json!({
+                    "stride": "DebugFontVertex",
+                    "rate": 0,
+                }),
+                ];
+            // Primitive and brush shaders
+            } else {
+                descriptors.push(json!({
+                    "stride": "PrimitiveInstance",
+                    "rate": 1,
+                }));
+            }
+            descriptors
+        };
+        pipeline_requirmenets.insert("attributeDescriptors", json!(attribute_descriptors));
+        pipeline_requirmenets.insert("descriptorPool", descriptor_pool);
+        pipeline_requirmenets.insert("descriptorSetLayouts", json!(descriptor_set_layouts));
+        pipeline_requirmenets.insert("vertexBufferDescriptors", json!(vertex_buffer_descriptors));
+        return Some(json!({
+            file_name.trim_right_matches(".vert"): pipeline_requirmenets
+        }));
+    }
+    None
+}
+fn compile_spirv(file_name_vector: Vec<String>, out_dir: &str) -> serde_json::value::Value {
+    let mut requirements = Vec::new();
+    for mut file_name in file_name_vector {
+        let file_path = Path::new(&out_dir).join(&file_name);
+        if let Some(req) = process_glsl_for_spirv(&file_path, &file_name) {
+            requirements.push(req);
+        }
+        file_name.push_str(".spv");
+        let spirv_file_path = Path::new(&out_dir).join(&file_name);
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        let mut glslang_cmd = Command::new(Path::new("./tools/glslangValidator"));
+        #[cfg(target_os = "windows")]
+        let mut glslang_cmd = Command::new(Path::new("./tools/glslangValidator.exe"));
+        glslang_cmd
+            .arg("-V")
+            .arg("-o")
+            .arg(&spirv_file_path)
+            .arg(&file_path);
+        if glslang_cmd
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit())
+            .status()
+            .unwrap()
+            .code()
+            .unwrap() != 0
+        {
+            println!("Error while compiling spirv: {:?}", file_name);
+            process::exit(1)
+        };
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        let mut spirv_val_cmd = Command::new(Path::new("./tools/spirv-val"));
+        #[cfg(target_os = "windows")]
+        let mut spirv_val_cmd = Command::new(Path::new("./tools/spirv-val.exe"));
+        spirv_val_cmd.arg(&spirv_file_path);
+        if spirv_val_cmd
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit())
+            .status()
+            .unwrap()
+            .code()
+            .unwrap() != 0
+        {
+            println!("Error while validating spirv shader: {:?}", file_name);
+            process::exit(1)
+        }
+    }
+    json!(requirements)
+}
+
+fn write_to_json(requriements: serde_json::value::Value, out_dir: &str) {
```

Comment:
This name is a bit misleading, because you use the name `write_json` before.
Maybe `write_json_to_file`?